features/dd-sdk-android-rum/src/main/kotlin/com/datadog/android/rum/internal/metric/interactiontonextview/InteractionToNextViewMetricResolver.kt

```diff
@@ -0,0 +1,116 @@
+        }
+        val lastPrevViewInteraction = resolveLastInteraction(viewId, currentViewCreatedTimestamp)
+        if (lastPrevViewInteraction != null) {
+            val difference = currentViewCreatedTimestamp - lastPrevViewInteraction.eventCreatedAtNanos
+            if (difference > 0) {
+                return difference
+            } else {
+                internalLogger.log(
+                    InternalLogger.Level.WARN,
+                    InternalLogger.Target.MAINTAINER,
+                    {
+                        "[ViewNetworkSettledMetric] The difference between the last interaction " +
+                            "and the current view is negative for viewId:$viewId"
+                    }
+                )
+                return null
+            }
+        }
+        internalLogger.log(
+            InternalLogger.Level.WARN,
+            InternalLogger.Target.MAINTAINER,
+            { "[ViewNetworkSettledMetric] No previous interaction found for this viewId:$viewId" }
+        )
+        return null
+    }
+
+    private fun resolveLastInteraction(viewId: String, currentViewCreatedTimestamp: Long): InternalInteractionContext? {
+        val currentViewIdIndex = lastViewCreatedTimestamps.keys.indexOf(viewId)
+        val previousViewId = lastViewCreatedTimestamps.keys.elementAtOrNull(currentViewIdIndex - 1)
+        if (previousViewId != null) {
+            lastInteractions[previousViewId]?.let {
+                val context = PreviousViewLastInteractionContext(
+                    it.actionType,
+                    it.eventCreatedAtNanos,
+                    currentViewCreatedTimestamp
+                )
+                if (lastInteractionIdentifier.validate(context)) {
+                    return it
+                }
+            }
+        }
+        return null
+    }
+
+    private fun purgeOldEntries() {
+        while (lastInteractions.entries.size > MAX_ENTRIES) {
+            @Suppress("UnsafeThirdPartyFunctionCall")
+            // we make sure the collection is never empty
+            lastInteractions.entries.remove(lastInteractions.entries.first())
+        }
+        while (lastViewCreatedTimestamps.size > MAX_ENTRIES) {
+            @Suppress("UnsafeThirdPartyFunctionCall")
+            // we make sure the collection is never empty
+            lastViewCreatedTimestamps.remove(lastViewCreatedTimestamps.keys.first())
+        }
+    }
+
+    @VisibleForTesting
+    internal fun lasInteractions(): Map<String, InternalInteractionContext> {
+        return lastInteractions
+    }
+
+    @VisibleForTesting
+    internal fun lastViewCreatedTimestamps(): Map<String, Long> {
+        return lastViewCreatedTimestamps
+    }
+
+    companion object {
+        // we need to keep at least 10 entries to be able to calculate the metric for consecutive views that
+        // are going to be created almost in the same time and will rely on the previous view interaction
+        private const val MAX_ENTRIES = 10
```

Comment:
I don't quite understand why "we need to keep at least 10 entries", can you elaborate? for example if we keep less than 10 entries, what's the consequence?

Reply:
Yeah like I answered to @ncreated question above is because we can always have `inactive` views that are still asking for the metric so we need to handle multiple views here. Also later we might need to handle `multiple active` views. I think this number is too big though so I will lower it to 3 entries for now.